### ISSUE TITLE
Maintenance: More simplification of `__init__.py`

### ISSUE THREAD

**Issue by hmaarrfk** [scikit-image/scikit-image#3155]
## Description
In my mind, `__init__.py` is very cluttered. Here is a breakdown of the lines of code in `__init__.py`.

Happy to do PRs for any of the suggestions that get approved.

| Lines | Purpose | Fraction of file | Alternative |
|-------|---------|--------|--------|
| 1-57  | Documentation   | large| Keep, this is great!  |
| 59, 65-66 | data_dir  | small  | ☑ [data_dir] Fixed in  #3161 |
| 68   | version  |   | Keep, this is great!  |
| 71-88   | python version  |  medium  | ☑ [python version] Fixed in #3160 |
| 91-131 | Test  |  large | ☑ [pytest] Fixed in #3164 |
| 134-178 | `__SKIMAGE_SETUP__` hack |  large | ☐ Addressed in #3158  |
| 181-192 | Numpy cross/docs | medium | ☑ [lookfor] Fixed in #3162 |

With refactoring, the file would look like:

```python
"""Docs
[...]
"""
__version__ = "0.15.dev0"

from .util.dtype import *
from .data import data_dir

from .util import lookfor
```

### [datadir] 
I suggest:
  - `from .data import data_dir`
  - Or just tell users to `from skimage.data import data_dir as skimge_datadir` as they are only likely to use this in the initial tutorial phase of their work.

### [python version] 
I think this should be removed.
Also, this problem shouldn't happen much most users will be installing scikit-image from pip/conda where 0.15 won't even exist in python2 repos.

Alternatively, it should point to `http://scikit-image.org/docs/dev/install.html` and not duplicate the instructions IMO.

### [pytest] 
Instead of having our own wrapper for pytest, just tell users to run the pytest command themselves.

From the terminal: `python -m pytest --pyargs skimage`
From a python console: `import pytest; pytest.main('--pyargs skimage')`

The reasoning is:
  1. It never worked before. See PR #3152.
  2. It was never tested before #3152.
  3. I don't think you should have wrappers for 1 liners.

### [lookfor] 
This one is tricky because it uses introspection.

Maybe move to `skimage.utils.docs` but it uses introspection, so the code would have to be rewritten too.

**Comment by stefanv** [comment]
I like this refactoring! We probably should keep 'test', though, as it is used by the wheel builder to test the packaged binary.

We can also look into runtests, as used by NumPy and SciPy.

**Comment by hmaarrfk** [comment]
Re [pytest]: It seems that the linux/OSX builds do not use the `skimage.test` function. They simply call `pytest` directly.
https://github.com/scikit-image/scikit-image-wheels/blob/master/config.sh#L25
Though, at the moment, I can't find what calls the `run_tests` functions defined above.

And so do the windows builds
https://github.com/scikit-image/scikit-image-wheels/blob/master/.appveyor.yml#L84

**Comment by hmaarrfk** [comment]
The omnibus PR #3147 has been broken up into many smaller PRs. 

This is a personal note that if/when they get merged, I need to check PR #3147 to ensure that all the functionality that I had there has been implemented.

**Comment by hmaarrfk** [comment]
I think this is fixed.
The `__SKIMAGE_SETUP__` seems to be an idiom shared by many projects using numpy's distutils.
